### DISCUSSION TITLE
Add Google Maps or Other Map services?

### DISCUSSION THREAD

**Discussion by pil0706** [teslamate-org/teslamate#1200] [Ideas]
I am living in S.Korea where OpenStreetMap is not precise nor Google maps.

I wanted to suggest add local maps within teslamate dashboard and let users type tile XYZ url somewhere under Web App's setting page.

I think this is a good spot?
![image](https://user-images.githubusercontent.com/4505669/104484868-21689e00-560d-11eb-9dc4-33d766accc4a.png)

if the box is empty, Teslamate serves default OSM maps.
if the box is filled with a valid tile XYZ then it serves that map tiles.

I searched JS files within teslamate and found out **teslamate** uses only one map and possibly hybrid map layer (satelllite & road networks) if that is allowed.

```js
function createMap(opts) {
  const map = new M(opts.elId != null ? `map_${opts.elId}` : "map", opts);

  const osm = new TileLayer(
    "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
    { maxZoom: 19 }
  );

  if (opts.enableHybridLayer) {
    const hybrid = new TileLayer(
      "http://{s}.google.com/vt/lyrs=s,h&x={x}&y={y}&z={z}",
      { maxZoom: 20, subdomains: ["mt0", "mt1", "mt2", "mt3"] }
    );

    new Control.Layers({ OSM: osm, Hybrid: hybrid }).addTo(map);
  }

  map.addLayer(osm);

  return map;
}
```

please consider my suggestion and i would like to see in next version update

**Comment by bloy123** [comment]
good idear

**Comment by eliluong** [comment]
If you login to the Grafana dashboard directly, you can edit the dashboards and change the map provider. For example, on the `Visited` dashboard, when you look at `Basemap layer`, you can change it from Open Street Map to something else.

![image](https://github.com/teslamate-org/teslamate/assets/15806938/b3668690-806c-4399-bbe7-0af91460ea98)

Here I used the `XYZ Tile layer` with [LocationIQ](https://docs.locationiq.com/docs/maps#leafletjs-and-other-libraries).

![image](https://github.com/teslamate-org/teslamate/assets/15806938/946192a3-2cae-4698-bf85-c2d3f3016fda)

You have to go to each dashboard that has a map and change it (eg Drive Details, Visited, etc), but this should work until someone has the time to look and add the feature.
